### DISCUSSION TITLE
중급프로젝트 최종 팀회고

### DISCUSSION THREAD

**Discussion by yeji0512** [codeit-bootcamp-frontend/2-Weekly-Mission#432] [회고]
중급 프로젝트 최종 팀 회고를 하고([회고 방식 상세보기](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/discussions/429#discussion-6130547)), 이전처럼 댓글 남겨주세요.
서로 고생 많았다고 다독여주는 시간, 그리고 아쉬웠던 부분은 회고할 수 있는 시간 되었으면 좋겠어요.  

중급 프로젝트 너무 너무 고생많으셨습니다 💜

**Comment by Chyyni** [comment]
## 4팀
### keep
- 프로젝트 시작 전, 컨벤션을 정하고 시작하여 조금 더 통일성있게 코드를 작성할 수 있었다.
- 트러블 슈팅을 문서화하여 모두 함께 공유한 점이 좋았다.
- 회의록 템플릿을 정해서 데일리 스크럼이 늘어지지 않고 진행돼서 좋았다.
- 팀원들 간 소통이 잘 되고, 의견을 나눌 수 있어서 좋았다.


### Problem
- 폴더 구조를 확실히 정하고 시작하지 않아서, 폴더 구조가 중구난방이다.
- 코드 리뷰를 활발히 하지 못한 점이 아쉬웠다.
- 코드 리뷰를 할 때, 다른 사람의 코드를 완전히 숙지하지 못한 채 넘어갔던 점이 아쉬웠다.
- pr의 크기를 작게 만들지 못하고, 크게 만들어 다른 팀원들이 pr내용을 이해하는 데 어려움을 겪었던 점이 아쉽다.
- 좀 더 활발한 토론을 했었으면 하는 아쉬움이 있다.
- 에러 처리를 충분히 하지 못한 점이 아쉽다.


### Try
- live share를 사용해보면 좋을 것 같다.
- 다음에는 내가 모르는 부분을 도전해보면 좋을 것 같다.
- 각자 고민하는 시간의 데드라인을 정하고, 그 시간이 지나면 바로 도움을 요청하도록 노력해야 할 것 같다.

**Comment by grapefruit13** [comment]
# 8팀

<img width="826" alt="image" src="https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/92169354/cbdf56f9-93c2-495e-9f25-f194cced2931">

**Comment by rladngus133** [comment]
# 9팀👊

## Keep
- 발표에서 최적화 , UX적인면을 많이 고려하고 노력했던 모습이 보여 좋았다. 
- 개발중에 각자 해결이 안 되는 부분이 있을때 빠른 피드백과 해결이 될 때 까지 다같이 고민해주는 모습이 좋았다.
- 팀원들이 서로를 믿고 의지해주어서 좋았다.
- 프로젝트 마무리와 발표준비를 위해서 설 연휴동안 군소리없이 시간을 내서 다같이 회의하는 모습이 좋았다.
- 무언가를 하자고 했을때 다 긍정적으로 하자고 해주었던 태도가 좋았다.
- 개발중 이슈 혹은 모두가 고려해야하는 의견을 던져줬을때 다 같이 빠지지않고 끝까지 고려하고 적극 검토하는 모습이 좋았다.

## Problem 
- 컨벤션을 조금더 명확하게 하였으면 했다. (esLint , Husky, Pritter 등등...)
- git branch push 할 시 계속 오류가 나와서 계속하여 삭제하고 다시 push 한게 아쉬웠다.

## Try
- 다음부터 작은부분까지 고려하여 컨벤션을 명확하게 설계하는것이 좋겠다.
- 오류 발생시 빠르게 하려고 삭제=>Re Push 하는게아닌 왜 오류가 발생하였는지 명확하게 파악하고 고쳐봐야겠다.

**Comment by juncastle97** [comment]
# 3팀
## Keep⭐️
 - 시간 내에 목표한 것을 완료했다.
 - 서로 부족한 부분을 잘 도와줬다.
 - 먼저 끝난 팀원들이 다른 부분도 도와서 작업했다.
 - 결정하는 거에 있어서 시간 낭비를 하지 않고 빠르게 결정했다.
## Problem💡
- 코드리뷰를 활발히 하지 못했다.
- merge 후에 서로의 코드를 확인하는 점이 부족했다.
- 컨벤션 통일이 잘 안 됐고, 폴더 구조 통일이 부족했다.
## Try🏃🏻
- 프로젝트 하면서 부족했던 부분 공부 다시 해보기
- 리팩토링 하기

3팀 모두 고생하셨습니다!

**Comment by muyahho** [comment]
![스크린샷 2024-02-14 114048](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/88885994/7a84c10a-a3df-4136-91f0-f250dc2078aa)

**Comment by 9ummy9** [comment]
# 7팀

## Keep 👌
   - 계획했던 기간 내에 프로젝트를 완성했다.
   - 모두가 성실하게 프로젝트에 참여했다. 

## Problem 😢
   - 자신이 맡은 기능을 설명하는 주석을 작성했으면 좋겠다. 
  
## Try 👍
   - 코드리뷰를 더 꼼꼼히 하고, PR 머지 규칙을 상세히 설정했으면 좋았을 것 같다.
   - 발표 내용을 모두가 논의하고 정리했으면 좋았을 것 같다.
   - 다음 프로젝트에서는 Github의 팀 레포지토리 기능도 활용해보고 싶다.

<br>

![2756563558AE7CDC11](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148167964/c6aa1b0c-ab87-4579-b3a8-c21f8ef726ca)


7팀 수고 많으셨습니다!

**Comment by mizign** [comment]
## Keep
- 목표한 기간내에 (설 이전에) 프로젝트를 완료했다.
- git 워크플로우를 다음에도 사용해보고 싶다.
- 소통/피드백이 빨랐다.
- 다음에도 husky로 린트룰을 맞추고 싶다.
- 작은 일이지만 트러블 슈팅 기록해둔게 좋았다.

## Problem
- 함께 해결한 이슈가 없어서 아쉬웠다. 
- 전체적인 설계를 생각해 볼 여유가 없었다. (상태관리)

## Try
- 다음 발표엔 시연을 동영상 형식으로 해야겠다.
- 인증 토큰 관리 플로우에 관해 더 고민해보고 싶다. 
- 코드리뷰에 도전해보고 싶다.

![itsme](https://github.com/codeit-bootcamp-frontend/2-Weekly-Mission/assets/148179726/0c513730-1a9d-4902-979f-cb06e1f107a0)

**Comment by ynmkim** [comment]
## 10팀

Keep

- 맡은 파트를 마지막까지 최선을 다하는 열정과 책임감이 멋졌다.
- 에러나 최신 업데이트 상황을 바로 공유하여 문제를 빠르게 해결하고자 노력하였다.

Problem

- 최소한의 공통 컴포넌트만 만들고 작업을 시작해서 중복되는 타입이나, fetch 함수들, 최소 단위 컴포넌트들이 많았다.
- 폴더 정의와 구조, 코드 스타일, 파일명 등의 규칙들이 정리 되지 않아서 제각각이었다.
- 일정 관리가 잘 되지 않았다.

Try

- 리패토링을 통해 재사용 가능한 코드는 분리하고 중복되는 코드들은 줄여나가야 할 것 같다.

**Comment by sung-eunji** [comment]
## keep
- 깃허브 프로그레스 관리가 좋았습니다.
- 브랜치 전략
- github을 활용한 문서화 -> 여러 군데에 왔다갔다 하지 않아도 돼서 보기 편함
- 깃헙 프로젝트 관리에 대한 To do ~done 등에 관한 작업단위의 정함이 프로젝트 진행 방향, 진도, 앞으로 해야할 일에 대한 부분 파악하기가 쉬웠습니다
- 팀원간 소통하거나 의견에 대한 조언을 구하거나, 문제해결을 위한 공유 라이브 쉐어에 대한 경험이 좋았습니다.
- 우리 깃허브 사용 진짜 잘 한 거 같죠?
- 우리 업무 분담 꽤 잘 한 거 같죠?


## Problem
- 마감기한이 필요한 주범입니다... 자꾸 욕심내다가 프로젝트에 기여를 못한 것 같아 죄송스러웠습니다.
- 코드 리뷰 받기만 하고 제가 한 게 몇 개 안되서 이거 또한 반성...
- 적정 마감 기한을 못 지킨 거 같아요. 앞으론 마감기한을 미리 정해두거나 하는 식으로 시간 관리를 하는 게 좋을 거 같아요.
- 너무 오래 고민하는...게 있었어요. 고민이 한 시간이 넘는다던가 하면 바로 팀원 등에 물어보는 걸 정해두는 게 좋을 거 같아요.
- pr리뷰에 대한 활발한 참여를 못한 부분이 죄송했습니다.
- 결국 다른부분에서 공통컴포넌트등 다른부분을 써야하는데 다시 살펴보고 이해해야 했던 부분으로 시간을 허비하는 경우도 있었고 리뷰에 적극 참여하여 미연에 이후 사용할 부분에 적용을 미리 생각해서 이후 사용이나 개선점을 제시하여 작업진행을 좀 더 부드럽게 하는게 좋았을거 같은 아쉬움이 있습니다.
- 파일등을 작성하거나 사용하는 과정에서 코드는 만들었으나 이것의 용도에 대한 이해 불충분으로 용도에 맞지않는 폴더에 넣거나 한 부분이 포로젝트 폴더구조에 혼란을 주었던거 같습니다.
- 코드 리뷰 할 에너지 부족
- 너무 널널했던 컨벤션(코드, 폴더 구조, 파일 등)
- 기술 스택 선정 근거 부족
- 요구 사항 분석 미흡

## Try
- 프로젝트 시작 전 전체 요구 사항 분석과 코드 컨벤션을 정하는 데 많은 시간과 리소스를 쓰자
- 코드 리뷰 스케쥴을 정하는 건 어떨까?
- 모르는 부분이나 다른 코드를 참고할때 적용에만 중점을 두지말고 뜯어보고 원리에 대한 이해 시간을 가지도록 노력하겠습니다
- 마감 기한에대한 좀더 분명한 가이드라인이 있는것은 어떠신지요
- 좀 더 다양한 라이브러리를 찾아보고 싶네요
- 코드 리뷰를 좀 더 꼼꼼히 해보고 싶네요(CLI 써보고 싶어요)
- Tailwind 말고 모듈이나 scss를 써보고 싶네요
